File: Hello_World_Tutorials/testing/c++/circ_sllist.cpp
/**
 * Created by Kalpathi Subramanian, 1/30/18
 * krs@uncc.edu
 */

#include "Bridges.h"
#include "CircSLelement.h"
#include "StudentInfo.h"

using namespace bridges;

int main (int argc, char **argv) {
	Bridges *bridges =  new Bridges(105, "YOUR_USER_ID", "YOUR_API_KEY");


	// create the linked list elements with
	// student data
	CircSLelement<StudentInfo> *students[] = {
		new CircSLelement<StudentInfo>(
			StudentInfo(
				"00000000000",
				"Gretel Chaney",
				"CS",
				"g.chaney@generated.com",
				"magenta",
				"blue",
				9.0
			), ""),
		new CircSLelement<StudentInfo>(
			StudentInfo(
				"00000000001",
				"Karol Soderman",
				"SIS",
				"k.soderman@generated.com",
				"magenta",
				"red",
				11.0
			), ""),
		new CircSLelement<StudentInfo>(
			StudentInfo(
				"00000000002",
				"Lamont Kyler",
				"BIO",
				"l.kyler@generated.com",
				"yellow",
				"green",
				12.0
			), ""),
		new CircSLelement<StudentInfo>(
			StudentInfo(
				"00000000003",
				"Gladys Serino",
				"CS",
				"g.serino@generated.com",
				"green",
				"magenta",
				9.0
			), ""),
		new CircSLelement<StudentInfo>(
			StudentInfo(
				"00000000004",
				"Starr Mcginn",
				"CS",
				"s.mcginn@generated.com",
				"red",
				"cyan",
				15.0
			), "")
	};

	CircSLelement<StudentInfo> *head =  students[0];
	CircSLelement<StudentInfo> *current =  head;

	int num_students = 5;
	for (int i = 1; i < num_students; i++) {
		current->setNext(students[i]);
		current = current->getNext();
	}
	//  point the last element to the first element,
	current->setNext(head);

	//  add visual attributes
	current = head;
	do {
		current->setLabel(current->getValue().getStudentLabel());
		current->getVisualizer()->setColor(current->getValue().getFavoriteColor());

		current->getLinkVisualizer(current->getNext())->setColor(current->getValue().getDislikeColor());
		current->getLinkVisualizer(current->getNext())->setThickness(current->getValue().getStudentCreditHours() * 0.5);

		current = current->getNext();
	}   while (current != head);


	// set data structure to point to head
	bridges->setDataStructure(head);
	// visualize the circular list
	bridges->visualize();
}

File: Hello_World_Tutorials/testing/python/circ_sllist.py
from Bridges.Bridges import *
from StudentInfo import *
from Bridges.CircSLelement import *

class CircSLtutorial:

    # create a bridges object with your credentials
    bridges = Bridges(5, "YOUR_USER_ID", "YOUR_API_KEY")

    students = []

    students.append(Student("00000000000",
                            "Gretel Chaney",
                            "CS",
                            "g.chaney@generated.com",
                            "magenta",
                            "blue",
                            9.0))

    students.append(Student("00000000001",
                            "Karol Soderman",
                            "SIS",
                            "k.soderman@generated.com",
                            "magenta",
                            "red",
                            11.0))

    students.append(Student("00000000002",
                            "Lamont Kyler",
                            "BIO",
                            "l.kyler@generated.com",
                            "yellow",
                            "green",
                            50.0))

    students.append(Student("00000000003",
                            "Gladys Serino",
                            "CS",
                            "g.serino@generated.com",
                            "green",
                            "magenta",
                            9.0))

    students.append(Student("00000000004",
                            "Starr Mcginn",
                            "CS",
                            "s.mcginn@generated.com",
                            "red",
                            "cyan",
                            15.0))


    head = CircSLelement(label = "", e = students[0])
    current = head

    for i in range(1, len(students)):
        current.set_next(CircSLelement(label = "", e = students[i]))

        #handle the last element
        if i is len(students)-1:
            #getting the last element
            current = current.get_next()

            # point the last element to the first element
            # so the list becomes circular
            current.set_next(head)
        # set the current element to be the next element
        current = current.get_next()

    current = head

    # add visual attributes
    for i in range(len(students)):

        current.set_label(current.get_value().getName())
        current.get_visualizer().set_color(current.get_value().getLikeColor())

        current.get_link_visualizer(current.get_next()).set_color(current.get_value().getDislikeColor())

        current.get_link_visualizer(current.get_next()).set_thickness(current.get_value().getCreditHours() *.03)

        current = current.get_next()

    bridges.set_data_structure(head)
    bridges.visualize()
